keps/sig-testing/2464-kubetest2-ci-migration/README.md

```diff
@@ -0,0 +1,743 @@
+
+    ```shell script
+    kubetest2 --test=exec – script.sh
+    ```
+
+- [`scenario=kubernetes_e2e`](https://github.com/kubernetes/test-infra/blob/2abb7ff26579325f6a335990003f665755c96d7a/scenarios/kubernetes_e2e.py) 
+ is used for setting up the environment for end-to-end jobs.
+    - It specifies it’s own set of flags to customize the setup all of which are deprecated and unused.
+    - It has a lot of functionality for setting up Kops clusters which will need to be ported over to be part of the kubetest2 kops deployer. They are currently being monitored in testgrid here: https://testgrid.k8s.io/kops-kubetest2#e2e-kubetest2
+
+    Most other usages in jobs are used to indirectly invoke kubetest, which will be replaced by their equivalent kubetest2 commands.
+
+    ```shell script
+    kubetest2 <deployer> --build --up --down --deployer-specific-setup-flags
+    ```
+- [`scenarios=kubernetes_build`](https://github.com/kubernetes/test-infra/blob/2abb7ff26579325f6a335990003f665755c96d7a/scenarios/kubernetes_build.py) which is primarily used for pushing CI builds to GCS.
+    This is where we are also using the legacy push-build scripts (in addition to kubetest --stage).
+    [These will be replaced by KREL.](https://github.com/kubernetes/release/issues/1711)
+
+    Optionally they can also be migrated to use the kubetest2 KREL integration.
+    ```shell script
+    kubetest2 noop --build --stage=gs://bucket
+    ```
+
+    Additionally, there is exactly one usage of `scenario=canarypush` which can be replaced by an equivalent script and migrated to using kubetest2 exec tester.
+
+#### Managing kubetest2 dependency
+
+- Add a script to kubekins-e2e to fetch pre-built latest kubetest2 binaries (stored in GCS) as part of each job
+- Distribute a tagged version of kubetest2 binary for each release and add to variants.yaml for kubekins-e2e
+- Publish a kubetest2 specific images (optional)
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+### Graduation Criteria
+
+- This will be declared **alpha** when:
+    - GCE jobs have corresponding kubetest2 shadow jobs on k/k master
+    - Noop support is added for node deployer and build
+
+
+- This will be declared **beta** when:
```

Comment:
I sorta wanted to suggest this in alpha but leave it to you
```suggestion
- This will be declared **beta** when:
    - A guide for migration from kubetest to kubetest2 exists
```